client/service/src/builder.rs

```diff
@@ -1495,3 +1241,384 @@ ServiceBuilder<
 		self.build_common()
 	}
 }
+
+// Inform the tx pool about imported and finalized blocks.
+async fn txpool_notifications<TBl, TBackend, TExec, TRtApi, TExPool>(
+	client: Arc<Client<TBackend, TExec, TBl, TRtApi>>,
+	txpool: Arc<TExPool>
+)
+	where
+		TBl: BlockT,
+		TExec: CallExecutor<TBl>,
+		Client<TBackend, TExec, TBl, TRtApi>: ProvideRuntimeApi<TBl>,
+		TExPool: MaintainedTransactionPool<Block=TBl>,
+{
+	let import_stream = client.import_notification_stream().map(Into::into).fuse();
+	let finality_stream = client.finality_notification_stream()
+		.map(Into::into)
+		.fuse();
+
+	futures::stream::select(import_stream, finality_stream)
+		.for_each(|evt| txpool.maintain(evt))
+		.await
+}
+
+async fn offchain_notifications<TBl, TBackend, TExec, TRtApi>(
```

Comment:
I would have assumed that this is moved to the offchain crate.

Reply:
We'd need to move `Client` out of `sc-service` again in order to do that. @seunlanlege what are your thoughts on this?

Reply:
Wait, no, we can just use the client api :facepalm: